src/subdomains/payout/strategies/prepare/impl/defichain.strategy.ts

```diff
@@ -19,28 +22,11 @@ export class DeFiChainStrategy extends PrepareStrategy {
     super();
   }
 
-  async preparePayout(order: PayoutOrder): Promise<void> {
-    try {
-      const { asset, amount, context } = order;
-
-      if (!(await this.defichainService.isHealthy(context))) return;
-
-      const destinationAddress = this.defichainService.getWalletAddress(context);
-      const request = { asset, amount, destinationAddress };
-
-      const transferTxId = await this.dexService.transferLiquidity(request);
-
-      order.pendingPreparation(transferTxId);
-    } catch (e) {
-      console.error(`Error in transferring liquidity for payout order. Order ID: ${order.id}`, e);
-      return;
-    }
+  async preparePayout(orders: PayoutOrder[]): Promise<void> {
+    const groups = PayoutUtils.groupOrdersByContext(orders);
 
-    try {
-      await this.payoutOrderRepo.save(order);
-    } catch (e) {
-      // db failure case, internal transfer - just logging is sufficient
-      console.error(`Error in saving liquidity transfer txId to payout order. Order ID: ${order.id}`, e);
+    for (const [context, group] of [...groups.entries()]) {
```

Comment:
`[...groups.entries()]` is not required (use `groups.entries()`)

Reply:
totally right, just Iterable will work